packages/mermaid/src/diagrams/xychart/chartBuilder/components/axis/BaseAxis.ts

```diff
@@ -0,0 +1,342 @@
+          path: `M ${x},${this.getScaleValue(tick)} L ${
+            x - this.axisConfig.tickLength
+          },${this.getScaleValue(tick)}`,
+          strokeFill: this.axisThemeConfig.tickColor,
+          strokeWidth: this.axisConfig.tickWidth,
+        })),
+      });
+    }
+    if (this.showTitle) {
+      drawableElement.push({
+        type: 'text',
+        groupTexts: ['left-axis', 'title'],
+        data: [
+          {
+            text: this.title,
+            x: this.boundingRect.x + this.axisConfig.titlePadding,
+            y: this.range[0] + (this.range[1] - this.range[0]) / 2,
+            fill: this.axisThemeConfig.titleColor,
+            fontSize: this.axisConfig.titleFontSize,
+            rotation: 270,
+            verticalPos: 'center',
+            horizontalPos: 'top',
+          },
+        ],
+      });
+    }
+    return drawableElement;
+  }
+  private getDrawaableElementsForBottomAxis(): DrawableElem[] {
+    const drawableElement: DrawableElem[] = [];
+    if (this.showLabel) {
+      drawableElement.push({
+        type: 'text',
+        groupTexts: ['bottom-axis', 'label'],
+        data: this.getTickValues().map((tick) => ({
+          text: tick.toString(),
+          x: this.getScaleValue(tick),
+          y: this.boundingRect.y + this.axisConfig.lablePadding + this.axisConfig.tickLength,
+          fill: this.axisThemeConfig.labelColor,
+          fontSize: this.axisConfig.labelFontSize,
+          rotation: 0,
+          verticalPos: 'center',
+          horizontalPos: 'top',
+        })),
+      });
+    }
+    if (this.showTick) {
+      const y = this.boundingRect.y;
+      drawableElement.push({
+        type: 'path',
+        groupTexts: ['bottom-axis', 'ticks'],
+        data: this.getTickValues().map((tick) => ({
+          path: `M ${this.getScaleValue(tick)},${y} L ${this.getScaleValue(tick)},${
+            y + this.axisConfig.tickLength
+          }`,
+          strokeFill: this.axisThemeConfig.tickColor,
+          strokeWidth: this.axisConfig.tickWidth,
+        })),
+      });
+    }
+    if (this.showTitle) {
+      drawableElement.push({
+        type: 'text',
+        groupTexts: ['bottom-axis', 'title'],
+        data: [
+          {
+            text: this.title,
+            x: this.range[0] + (this.range[1] - this.range[0]) / 2,
+            y: this.boundingRect.y + this.boundingRect.height - this.axisConfig.titlePadding,
+            fill: this.axisThemeConfig.titleColor,
+            fontSize: this.axisConfig.titleFontSize,
+            rotation: 0,
+            verticalPos: 'center',
+            horizontalPos: 'bottom',
+          },
+        ],
+      });
+    }
+    return drawableElement;
+  }
+  private getDrawaableElementsForTopAxis(): DrawableElem[] {
+    const drawableElement: DrawableElem[] = [];
+    if (this.showLabel) {
+      drawableElement.push({
+        type: 'text',
+        groupTexts: ['bottom-axis', 'label'],
```

Comment:
Why does the topAxis function have `bottom-axis` value?

Reply:
Copied issue, fixed

Reply:
Fixed